pkg/volume/csi/csi_client.go

```diff
@@ -61,45 +62,15 @@ type csiClient interface {
 
 // csiClient encapsulates all csi-plugin methods
 type csiDriverClient struct {
-	network          string
-	addr             string
-	conn             *grpc.ClientConn
-	idClient         csipb.IdentityClient
-	nodeClient       csipb.NodeClient
-	ctrlClient       csipb.ControllerClient
-	versionAsserted  bool
-	versionSupported bool
-	publishAsserted  bool
-	publishCapable   bool
+	driverName string
+	nodeClient csipb.NodeClient
 }
 
-func newCsiDriverClient(network, addr string) *csiDriverClient {
-	return &csiDriverClient{network: network, addr: addr}
-}
-
-// assertConnection ensures a valid connection has been established
-// if not, it creates a new connection and associated clients
-func (c *csiDriverClient) assertConnection() error {
-	if c.conn == nil {
-		conn, err := grpc.Dial(
-			c.addr,
-			grpc.WithInsecure(),
-			grpc.WithDialer(func(target string, timeout time.Duration) (net.Conn, error) {
-				return net.Dial(c.network, target)
-			}),
-		)
-		if err != nil {
-			return err
-		}
-		c.conn = conn
-		c.idClient = csipb.NewIdentityClient(conn)
-		c.nodeClient = csipb.NewNodeClient(conn)
-		c.ctrlClient = csipb.NewControllerClient(conn)
-
-		// set supported version
-	}
+var _ csiClient = &csiDriverClient{}
 
-	return nil
+func newCsiDriverClient(driverName string) *csiDriverClient {
```

Comment:
nit: return the interface (`csiClient`) not the struct

Reply:
You are right.  I missed that.